### PR TITLE
Add queries for smartseq3 tables, can also be used for other protocols

### PR DESCRIPTION
**Pull request by aanil**

### CODE REVIEW

File: genologics_sql/queries.py [hunk 1]
```diff
@@ -221,3 +221,73 @@ def get_children_processes(session, parent_process, ptypes, sample=None, orderby
...
+    :returns: List of sample udf values
+    """
+    query = f"select udfname, udfvalue, udfunitlabel \
+              from sample_udf_view \
+              where sampleid={sampleid} and udfname in {tuple(udf_list)};"
+
+    return session.execute(text(query)).all()
+
+
+def get_udfs_from_step(session, sampleid, stepid, udf_list):
```
Comment (alneberg) [inline genologics_sql/queries.py]: Change the name slightly so that it mentions the sample as well.

File: genologics_sql/queries.py [hunk 2]
```diff
@@ -221,3 +221,73 @@ def get_children_processes(session, parent_process, ptypes, sample=None, orderby
...
+              where st.stageid=stg.stageid and stg.membershipid=wfs.sectionid \
+              and wfs.workflowid=lwf.workflowid and wfs.protocolid=lpt.protocolid \
+              and st.artifactid=asm.artifactid and asm.processid=sa.processid and st.completedbyid is null \
+              and lwf.workflowstatus='ACTIVE' and lwf.workflowname='{workflow}';"
+
+    return session.execute(text(query)).all()
+
+
+def get_udfs_from_sample(session, sampleid, udf_list):
+    """returns all values of samples udfs in the udf_list
```
Comment (alneberg) [inline genologics_sql/queries.py]: Mention that udfs from steps are not included.

File: genologics_sql/queries.py [hunk 3]
```diff
@@ -221,3 +221,73 @@ def get_children_processes(session, parent_process, ptypes, sample=None, orderby
...
+
+    :param session: the current SQLAlchemy session to the db
+    :param sampleid: sampleid from sample table
+    :param udf_list: the list of udf names to be searched
+    :returns: List of sample udf values
+    """
+    query = f"select udfname, udfvalue, udfunitlabel \
```
Comment (alneberg) [inline genologics_sql/queries.py:278]: Maybe have it return all udfs if the udf_list is empty?

Comment (aanil) [reply genologics_sql/queries.py:278]: Done!

File: genologics_sql/queries.py (hunk 1 above)
Comment (aanil) [reply genologics_sql/queries.py]: Done!

File: genologics_sql/queries.py (hunk 2 above)
Comment (aanil) [reply genologics_sql/queries.py]: Done!

File: genologics_sql/queries.py [hunk 4]
```diff
@@ -221,3 +221,78 @@ def get_children_processes(session, parent_process, ptypes, sample=None, orderby
...
+    return session.execute(text(query)).all()
+
+
+def get_udfs_from_sample(session, sampleid, udf_list):
+    """returns all values from the given list of udfs which are directly connected to a sample.
+        It does not return the udfs from protocol steps.
+
+    :param session: the current SQLAlchemy session to the db
+    :param sampleid: sampleid from sample table
+    :param udf_list: the list of udf names to be searched
```
Comment (alneberg) [inline genologics_sql/queries.py]:
```suggestion
    :param udf_list: the list of udf names to be searched. If the list is empty or None, all UDFs are returned.
```

**Review by alneberg — APPROVED** [review]
This will be nice!

**Pull request merged**